gateway/test/dapi_alpha_test.go

```diff
@@ -11,3 +14,757 @@ func (s *GatewayOpsTestSuite) TestDapiAlphaEnabled() {
 	})
 	requireRestSuccess(s.T(), resp)
 }
+
+func (s *GatewayOpsTestSuite) insertBinaryDocument() string {
+	docId := s.randomDocId()
+	resp := s.sendTestHttpRequest(&testHttpRequest{
+		Method: http.MethodPost,
+		Path: fmt.Sprintf(
+			"/v1.alpha/buckets/%s/scopes/%s/collections/%s/documents/%s/increment",
+			s.bucketName, s.scopeName, s.collectionName, docId,
+		),
+		Headers: map[string]string{
+			"Authorization": s.basicRestCreds,
+		},
+		Body: TEST_BINARY_CONTENT,
+	})
+	requireRestSuccess(s.T(), resp)
+	assertRestValidEtag(s.T(), resp)
+	assertRestValidMutationToken(s.T(), resp, s.bucketName)
+
+	return docId
+}
+
+func (s *GatewayOpsTestSuite) TestAppendToDocument() {
+	s.Run("Basic", func() {
+		docId := s.insertBinaryDocument()
+
+		// Append to the document
+		resp := s.sendTestHttpRequest(&testHttpRequest{
+			Method: http.MethodPost,
+			Path: fmt.Sprintf(
+				"/v1.alpha/buckets/%s/scopes/%s/collections/%s/documents/%s/append",
+				s.bucketName, s.scopeName, s.collectionName, docId,
+			),
+			Headers: map[string]string{
+				"Authorization": s.basicRestCreds,
+			},
+			Body: TEST_APPEND_CONTENT,
+		})
+		requireRestSuccess(s.T(), resp)
+		assertRestValidEtag(s.T(), resp)
+		assertRestValidMutationToken(s.T(), resp, s.bucketName)
+
+		// Verify the key-value is present
+		resp = s.sendTestHttpRequest(&testHttpRequest{
```

Comment:
This should reuse the existing validation methods that are built into the test framework (see the POST/GET/PUT tests in dapi_crud_test.go).